File: app/src/main/java/dev/ragnarok/fenrir/api/model/longpoll/AbsLongpollEvent.java
package dev.ragnarok.fenrir.api.model.longpoll;

public class AbsLongpollEvent {

    public static final int ACTION_MESSAGES_FLAGS_SET = 2;
    public static final int ACTION_MESSAGES_FLAGS_RESET = 3;
    public static final int ACTION_MESSAGE_ADDED = 4;
    public static final int ACTION_MESSAGE_EDITED = 5;
    public static final int ACTION_SET_INPUT_MESSAGES_AS_READ = 6;
    public static final int ACTION_SET_OUTPUT_MESSAGES_AS_READ = 7;
    public static final int ACTION_USER_IS_ONLINE = 8;
    public static final int ACTION_USER_IS_OFFLINE = 9;
    public static final int ACTION_MESSAGE_CHANGED = 18;
    public static final int ACTION_USER_WRITE_TEXT_IN_DIALOG = 63;
    public static final int ACTION_USER_WRITE_VOICE_IN_DIALOG = 64;
    public static final int ACTION_COUNTER_UNREAD_WAS_CHANGED = 80;

    public final int action;

    public AbsLongpollEvent(int action) {
        this.action = action;
    }

    public int getAction() {
        return action;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/api/model/longpoll/AbsLongpollEvent.kt
package dev.ragnarok.fenrir.api.model.longpoll

import dev.ragnarok.fenrir.api.adapters.LongpollUpdateDtoAdapter
import kotlinx.serialization.Serializable

@Serializable(with = LongpollUpdateDtoAdapter::class)
open class AbsLongpollEvent(val action: Int) {
    companion object {
        const val ACTION_MESSAGES_FLAGS_SET = 2
        const val ACTION_MESSAGES_FLAGS_RESET = 3
        const val ACTION_MESSAGE_ADDED = 4
        const val ACTION_MESSAGE_EDITED = 5
        const val ACTION_SET_INPUT_MESSAGES_AS_READ = 6
        const val ACTION_SET_OUTPUT_MESSAGES_AS_READ = 7
        const val ACTION_MESSAGE_CHANGED = 18
        const val ACTION_USER_TYPING_TEXT_IN_DIALOG = 63
        const val ACTION_USER_RECORDING_VOICE_IN_DIALOG = 64
        const val ACTION_USER_UPLOADING_PHOTO_IN_DIALOG = 65
        const val ACTION_USER_UPLOADING_VIDEO_IN_DIALOG = 66
        const val ACTION_USER_UPLOADING_FILE_IN_DIALOG = 67
        const val ACTION_COUNTER_UNREAD_WAS_CHANGED = 80
        const val ACTION_MESSAGE_REACTION_CHANGE = 601
    }
}
